tools/run_tests/jobset.py

```diff
@@ -211,6 +223,8 @@ def state(self, update_cache):
       # implemented efficiently. This is an experiment to workaround the issue by making sure
       # results.xml file is small enough.
       filtered_stdout = filtered_stdout[-128:]
+      self.result.message = filtered_stdout
+      self.result.elapsed_time = elapsed
       if self._xml_test is not None:
```

Comment:
After this change, there will be two independent ways to report the test result: using the JobResult class you've created and using the _xml_test element.
I think that's fine for now, but eventually, we should probably refactor the code so that the report.xml file is built from the JobResult class instances (I am not a big fan of the way we currently build the xml report directly from Job class' method). What do you think?